File: fastapi-ca/dataset/infra/db_models/dataset.py
from datetime import datetime
from sqlalchemy import String, Text, Float
from sqlalchemy.orm import Mapped, mapped_column
from database import Base

class Dataset(Base):
    __tablename__ = "Dataset"

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    project_id: Mapped[str] = mapped_column(String(36), nullable=False)
    flow_id: Mapped[str | None] = mapped_column(String(36), nullable=True)
    name: Mapped[str] = mapped_column(Text, nullable=False)
    size: Mapped[float] = mapped_column(Float, nullable=False)
    path: Mapped[str] = mapped_column(Text, nullable=False)

File: frontend/src/types/dataset.ts
export interface Dataset {
    id: string
    project_id: string
    flow_id: string | null
    name: string
    size: number
    path: string
  }
  
  export interface CreateDatasetBody {
    project_id: string
    flow_id?: string | null
    name: string
    size: number
    path: string
  }
  
  export interface UpdateDatasetBody {
    id: string
    flow_id: string | null
  }
  
  export interface GetDatasetsResponse {
    total_count: number
    page: number
    datasets: Dataset[]
  }
  
  